frontend/src/client/components/ExploreRentals.tsx

```diff
@@ -1,169 +1,69 @@
-import React, { useState, useEffect } from "react";
-import { NftWithMetadata, Nft } from "../../../types/nftTypes.js";
-import { ListingPanel } from "./ListingPanel";
-import { Popup } from "./Popup";
-import { RentDetails } from "./RentDetails";
-import { useMoralisQuery } from "react-moralis";
-import DatePicker from "react-datepicker";
-import "react-datepicker/dist/react-datepicker.css";
-import { mergeNftsWithMetadata } from "../lib/fetchNft";
+import React, { useEffect, useState } from "react";
+import { useAccount, useConnect } from "wagmi";
+import { Filter, FilterTypes, QueryFilter, QueryFilterTypes } from "../../../types/queryTypes.js";
+import { PageTypes } from "../../../types/types.js";
+import PaginatedNFTs from "../components/PaginatedNFTsPanel";
 
 export const ExploreRentals = () => {
-  const [dateFilterVisible, setDateFilterVisible] = useState(false);
-  const [dateRange, setDateRange] = useState<[null | Date, null | Date]>([null, null]);
-  const [startDate, endDate] = dateRange;
-  const [costFilterVisible, setCostFilterVisible] = useState(false);
-  const [startCost, setStartCost] = useState<null | number>(null);
-  const [endCost, setEndCost] = useState<null | number>(null);
-
-  const { data: rawNftListings, error, isLoading } = useMoralisQuery("Listing", query => query.limit(8), [], {});
-  const [nfts, setNfts] = useState<NftWithMetadata[]>([]);
-
-  const [selectedNft, setSelectedNft] = useState<NftWithMetadata | null>(null);
+  const { data: accountData } = useAccount();
+  const { isConnected, isDisconnected } = useConnect();
+  const [queryFilterList, setQueryFilterList] = useState<QueryFilter[]>([]);
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    const nftListings: Nft[] = rawNftListings.map(nft => {
-      return {
-        listing: nft.attributes.listing,
-        specification: nft.attributes.nftSpecification,
-      };
-    });
-    mergeNftsWithMetadata(nftListings).then(nftsWithMetadata => setNfts(nftsWithMetadata));
-  }, [rawNftListings]);
-
-  if (isLoading) {
-    return <div>Loading</div>;
-  }
-  if (error) {
-    return <div>ERROR {error}</div>;
-  }
-
-  const renderDateFilter = () => {
-    let text = "Filter by date";
-
-    if (startDate) {
-      text = "Filtering from " + startDate.toLocaleDateString() + " - ";
-      if (endDate) {
-        text += endDate.toLocaleDateString();
-      }
+    if ((isConnected && typeof accountData !== "undefined") || isDisconnected) {
```

Comment:
((isConnected && typeof accountData !== "undefined") || isDisconnected) ?

Reply:
If wallet is connected then at firstly we get accountData as undefined and then accountData is updated via hooks. So if the wallet is connected and accountData is available only then pass the query filters and load the listings.

Otherwise, If wallet is disconnected then load all the listings. The User will later have to connect to wallet in order to rent them.